src/Symfony/Component/Messenger/Command/FailedMessagesShowCommand.php

```diff
@@ -0,0 +1,136 @@
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Messenger\Stamp\SentToFailedTransportStamp;
+use Symfony\Component\Messenger\Transport\Receiver\ListableReceiverInterface;
+use Symfony\Component\Messenger\Transport\Receiver\MessageCountAwareInterface;
+
+/**
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ *
+ * @experimental in 4.3
+ */
+class FailedMessagesShowCommand extends AbstractFailedMessagesCommand
+{
+    protected static $defaultName = 'messenger:failed:show';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure(): void
+    {
+        $this
+            ->setDefinition([
+                new InputArgument('id', InputArgument::OPTIONAL, 'Specific message id to show'),
+                new InputOption('max', null, InputOption::VALUE_REQUIRED, 'Maximum number of messages to list', 50),
+            ])
+            ->setDescription('Shows one or more messages from the failure transport.')
+            ->setHelp(<<<'EOF'
+The <info>%command.name%</info> shows message that are waiting in the failure transport.
+
+    <info>php %command.full_name%</info>
+
+Or look at a specific message by its id:
+
+    <info>php %command.full_name% {id}</info>
+EOF
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output);
+
+        $receiver = $this->getReceiver();
+        if ($receiver instanceof MessageCountAwareInterface) {
+            if (1 === $receiver->getMessageCount()) {
+                $io->writeln('There is <comment>1</comment> message waiting in the failed transport.');
+            } else {
+                $io->writeln(sprintf('There are <comment>%d</comment> messages waiting in the failed transport.', $receiver->getMessageCount()));
+            }
+        }
+
+        if (!$receiver instanceof ListableReceiverInterface) {
+            throw new RuntimeException(sprintf('The "%s" receiver does not support listing or showing specific messages.', $this->getReceiverName()));
+        }
+
+        if (null === $id = $input->getArgument('id')) {
+            $this->listMessages($io, $input->getOption('max'));
+        } else {
+            $this->showMessage($id, $io);
+        }
+    }
+
+    private function listMessages(SymfonyStyle $io, int $max)
+    {
+        /** @var ListableReceiverInterface $receiver */
+        $receiver = $this->getReceiver();
+        $envelopes = $receiver->all($max);
+
+        $rows = [];
+        foreach ($envelopes as $envelope) {
+            /** @var SentToFailedTransportStamp $sentToFailedTransportStamp */
```

Comment:
shouldn't all occurences of "FailedTransport" be replaced by "FailureTransport" in this PR (with or without a space between the words, case insensitive :) )